bin/opssh

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+# vim: expandtab:tabstop=4:shiftwidth=4
+
+import argparse
+import ansibleutil
+import traceback
+import sys
+import os
+import re
+import tempfile
+import time
+import subprocess
+
+DEFAULT_PSSH_PAR=200
+PSSH = '/usr/bin/pssh'
+
+class Opssh(object):
+    def __init__(self):
+        self.file_path = os.path.join(os.path.dirname(os.path.realpath(__file__)))
+        self.ansible = ansibleutil.AnsibleUtil()
+
+        self.parse_cli_args()
+
+        if self.args.list_host_types:
+            self.ansible.print_host_types()
+            return
+
+        if self.args.env and \
+           self.args.host_type and \
+           self.args.command:
+            retval = self.run_pssh()
+            if retval != 0:
+                raise ValueError("ansible run failed")
+
+            return
+
+        # If it makes it here, we weren't able to determine what they wanted to do
+        raise ValueError("Invalid combination of arguments")
+
+    def run_pssh(self):
+        """Actually run the pssh command based off of the supplied options
+        """
+
+        # Default set of options
+        pssh_args = [PSSH, '-i', '-t', '0', '-p', str(self.args.par)]
+
+        if self.args.outdir:
+            pssh_args.append("--outdir='%s'" % self.args.outdir)
+
+        if self.args.errdir:
+            pssh_args.append("--errdir='%s'" % self.args.errdir)
+
+        hosts = self.ansible.get_host_list(self.args.host_type, self.args.env)
+        with tempfile.NamedTemporaryFile(prefix='opssh-', delete=True) as f:
+            for h in hosts:
+                f.write('root@' + h + os.linesep)
```

Comment:
Hard coded to the user root.  pssh does take a username.  Kind of like the option of doing this as non-root.
  -l USER, --user=USER  username (OPTIONAL)